### ISSUE TITLE
[RFC] Random background and the birthday problem

### ISSUE THREAD

**Issue by KoviRobi** [rycee/home-manager#923]
Hi,

I was thinking about altering the random background service, because currently it suffers from the [birthday problem](https://en.wikipedia.org/wiki/Birthday_problem), i.e. it takes a long time for you to see _every picture_, but a short time for you to start seeing ones you have already seen. My thinking was to instead randomize the the set of images at the start, then to consume images from that list until it is empty, then randomize it again.

I think this could be achieved with something like find, shuf, and tail, and will hopefully get around to hacking something up, just currently busy until the next fortnight so I thought I'd just put this out in case anybody has any comments/suggestions.

Thanks,
Robert

**Comment by rycee** [comment]
Thanks for the report! But adjusting the randomization algorithm seems better suited for the [feh](https://github.com/derf/feh) project since that's what the random-background module uses. Alternatively, the module could use a different background setting program if one exists with better randomization.

**Comment by KoviRobi** [comment]
I see that you don't want a random shell script that I have written to be in your repo, and be responsible for its maintenance. I did report it to [feh](https://github.com/derf/feh/issues/495), but I also made my script, which I use in with an `imports` field for home-manager (https://github.com/KoviRobi/feh-random-background). That's good enough for me, I feel like putting that logic inside feh would add unnecessary complexity to feh (though I did suggest how it could be done should somebody wish to).
